domain/snyk/scanner/scanner.go

```diff
@@ -263,27 +266,26 @@ func (sc *DelegatingConcurrentScanner) Scan(
 
 	sc.scanNotifier.SendInProgress(folderPath)
 	gitCheckoutHandler := vcs.NewCheckoutHandler()
-	defer func() {
-		if gitCheckoutHandler.CleanupFunc() != nil {
-			logger.Debug().Msg("Calling cleanup func for base folder")
-			gitCheckoutHandler.CleanupFunc()()
-		}
-	}()
+
 	waitGroup := &sync.WaitGroup{}
+	referenceBranchScanWaitGroup := &sync.WaitGroup{}
 	for _, scanner := range sc.scanners {
 		if scanner.IsEnabled() {
 			waitGroup.Add(1)
```

Comment:
I'd consider moving this into the go routine, as it's part of the scanner start logic.